docs/docs/subscription.fsx

```diff
@@ -23,61 +23,162 @@ We can setup forwarding of those events to be processed by our `update` function
+                    1000
+            { new IDisposable with
+                member _.Dispose() = JS.clearInterval intervalId }
+        start
+
+    let subscribe model =
+        [ ["timer"], timer Tick ]
+
+    Program.mkSimple init update (fun model _ -> printf "%A\n" model)
+    |> Program.withSubscription subscribe
+    |> Program.run
+
+
+(**
+`subscribe` answers the question: "Which subscriptions should be running?" It has the current program state, `model`, to use for decisions.
+When the model changes, `subscribe` is called. Elmish then starts or stops subscriptions to match what should be running.
+
+A subscription has an ID, `["timer"]` here, and a start function. The ID needs to be unique within that page.
+You're probably wondering why it is a list. This allows you to include dependencies. Later we will use this to make the timer change intervals.
 
-let timer initial =
-    let sub dispatch =
-        JS.setInterval
-            (fun _ ->
-                dispatch (Tick DateTime.Now)
-            )
-            1000
-            |> ignore
-    Cmd.ofSub sub
+Here, the timer subscription is always returned from `subscribe`, so it will stay running as long as the program is running.
+
+Let's look at an example where the timer can be turned off. First we add the field `enabled`.
+*)
+
+module ToggleTimer =
+
+    type Model =
+        {
+            current : DateTime
+            enabled : bool
+        }
 
-Program.mkSimple init update (fun model _ -> printf "%A\n" model)
-|> Program.withSubscription timer
-|> Program.run
+    type Msg =
+        | Tick of DateTime
 
+    let init () =
+        {
+            current = DateTime.MinValue
+            enabled = true
+        }
 
 (**
-In this example program initialization will call our subscriber (once) with inital `Model` state, passing the `dispatch` function to be called whenever an event occurs.
-However, any time you need to issue a message (for example from a callback) you can use `Cmd.ofSub`.
+`update` and `timer` are the same as before.
+*)
+    let update msg model =
+        match msg with
+        | Tick current ->
+            { model with
+                current = current
+            }
+
+    let timer onTick =
+        let start dispatch =
+            let intervalId = 
+                JS.setInterval
+                    (fun _ -> dispatch (onTick DateTime.Now))
+                    1000
+            { new IDisposable with
+                member _.Dispose() = JS.clearInterval intervalId }
+        start
+
+(**
+Next, change the subscribe function to check `enabled` before including the timer subscription.
+*)
+    let subscribe model =
+        [ if model.enabled then
+            ["timer"], timer Tick ]
+
+    Program.mkSimple init update (fun model _ -> printf "%A\n" model)
+    |> Program.withSubscription subscribe
+    |> Program.run
+
+(**
+All that's left is to add user interaction to change `enabled`. The timer will stop or start accordingly.
```

Comment:
> All that's left is to add user interaction to change `enabled`

I don't really like the "All that's left" because it seems to imply this is simple to do but if the user don't know how to do it he could be frustrated.

Reply:
I also think it would be better to show how to add that user interaction. 

We can using the React DSL for that and adding a mention that user is not limited to HTML interaction and here we just use it for simplicity.

Reply:
Personally, my main usage of the fsx is to have intellisense/compiler assistance so that I don't make typo in the API used.

Printing to console was how Eugene wrote the initial documentation, but personally I always found it difficult to understand because interactivity is limited with console. In general, I think HTML/React is "easy" enough to understand for our usage here.

Reply:
Sounds good. I'll say at the top that the examples use Fable.Elmish.React, and link to its doc. So they have a starting place for that.

Reply:
I ended up just including `open Feliz` above the `view` function. So it's clear what is being used.